python/mujinplanningclient/realtimerobotplanningclient.py

```diff
@@ -386,1649 +740,3318 @@ def GetTransform(self, targetname, connectedBodyName='', linkName='', geometryNa
+        if locationCollisionInfos is not None:
+            taskparameters['locationCollisionInfos'] = locationCollisionInfos
         taskparameters.update(kwargs)
         return self.ExecuteCommand(taskparameters, timeout=timeout)
 
-    def SetRobotBridgeIOVariables(self, iovalues, robotname=None, timeout=10, **kwargs):
-        """Sets a set of IO variables in the robot bridge.
+    def SetRobotBridgeIOVariables(self, iovalues, robotname=None, timeout=10, dynamicEnvironmentState=None, debuglevel=None, unit='mm', toolname=None, robotBridgeConnectionInfo=None, locationCollisionInfos=None, forceasync=None, **kwargs):
+        # type: (Any, Optional[str], float, Optional[types.SetRobotBridgeIOVariablesParametersDynamicEnvironmentState], Optional[int], str, Optional[str], Optional[types.SetRobotBridgeIOVariablesParametersRobotBridgeConnectionInfo], Optional[list[types.SetRobotBridgeIOVariablesParametersLocationCollisionInfosArrayElement]], Optional[bool], Optional[Any]) -> Optional[Any]
+        """
+        Sets a set of IO variables in the robot bridge.
         This should not lock self.env since it can happen during the runtime of a task and lock out other functions waiting in the queue.
 
         Args:
-            iovalues:
-            robotname (str, optional): Name of the robot
-            timeout (float, optional): Time in seconds after which the command is assumed to have failed. (Default: 10)
-            unit (str, optional): The unit of the given values. (Default: 'mm')
-            toolname (str, optional): Name of the manipulator. Defaults to currently selected tool
-            robotBridgeConnectionInfo (dict, optional): Information to set up a client to the robot bridge.
-            locationCollisionInfos (dict, optional): List of external collision IOs to be computed and sent in realtime.
-            forceasync (bool, optional):
+            iovalues: ioValue names with expected values.
+            robotname: Name of the robot (Default: None)
+            timeout: Time in seconds after which the command is assumed to have failed. (Default: 10)
+            dynamicEnvironmentState: Dynamic environment state that allows the user to set/create objects in a particular state dynamically. (Default: None)
+            debuglevel: Sets the debug level for the planning logs. For development. 3=INFO, 4=DEBUG, 5=VERBOSE. (Default: None)
+            unit: The unit of the given values. (Default: 'mm')
+            toolname: Name of the manipulator. Defaults to currently selected tool (Default: None)
+            robotBridgeConnectionInfo: Information to set up a client to the robot bridge. (Default: None)
+            locationCollisionInfos: List of external collision IOs to be computed and sent in realtime. (Default: None)
+            forceasync: (Default: None)
         """
         taskparameters = {
             'command': 'SetRobotBridgeIOVariables',
-            'iovalues': list(iovalues)
-        }
+            'unit': unit,
```

Comment:
Why do we send this every time now?

Reply:
Fixed. https://github.com/mujin/mujinplanningclientpy/pull/45/commits/5a35c7ae48a1cf74368e101762d1c26035519029

We'd defaulted to sending it, but turns on that only 9 out of 53 sends made any sense.